docs-raw/site/docs/fabric-gateway-features.md

```diff
@@ -401,6 +402,31 @@ spec:
           body: '{"title": "Service down for maintenance", "status": 503}'
 ```
 
+### Custom Filter Configuration
+
+In some situations it's desirable to have the ability to add extra [Skipper filters](https://opensource.zalando.com/skipper/reference/filters/) into a gateway managed ingress which aren't explicitly modeled in the Fabric Gateway Custom Resource Definition. To this end, the gateway operator allows the pass-through of an allow-list of filters which will be applied to all proxy routes managed by the gateway operator. This is achieved through the use of an annotation on a gateway resource as outlined in the below example. As can be seen in the example, additional filters need to be populated in an annotation named: `fabric/additional-filters` and chained (in the usual [Skipper manner](https://opensource.zalando.com/skipper/kubernetes/ingress-usage/#filters-and-predicates)) by using the `->` delimiter. 
+
+```yaml
+apiVersion: zalando.org/v1
+kind: FabricGateway
+metadata:
+  name: my-app-gateway
+  annotations:
+    fabric/additional-filters: fadeIn("1m") -> tee("https://some-other-backend.zalandoapis.com/api")
```

Comment:
Wondering if the prefix should be "zalando.org/", like the skipper ingress / stackset annotations

Reply:
Also they could be taken as a list, instead of in the skipper format with the ->

Reply:
we're not a top level project, so thought it might be a bit presumptious. Not sure if there are general guidelines around this though. :man_shrugging: I just wanted it to be clear from the name that it was specific to gateway functionality.

Reply:
> Also they could be taken as a list, instead of in the skipper format with the ->

Think better to stick close to the Skipper format. We link out to Skippers docs when describing

Reply:
Fair enough. I see the filters get appended so they happen after auth, makes sense. Do you think the name could have -append or something in it, to leave us with the option of another annotation that e.g. prepends filters?

Reply:
> Fair enough. I see the filters get appended so they happen after auth, makes sense. Do you think the name could have -append or something in it, to leave us with the option of another annotation that e.g. prepends filters?

Auth happens in the predicates. I hadn't thought about whether appending/prepending for filters was important. Are the filters executed in order? Suppose it's good that we are appending then, less chance of f&*kin up some of the filters that we add for functionality. Do you think we need to explicitly state that these are appended? `additional` kind of suggests that in anyways, but should we call it out in the docs?

Reply:
> should we call it out in the docs

Yeah I think so, wouldn't make a big deal of it just state that they are appended to the list of filters for the route.

> Are the filters executed in order

Yeah on the request side they get executed from first to last, then on the response side they're executed in reverse order if they make sense to be executed for responses https://opensource.zalando.com/skipper/reference/architecture/#routing-mechanism

So I think it's good to have them appended, just think we should indicate that in the name of the annotation so we could also have some other annotation for prepending filters if we needed to

Reply:
> Auth happens in the predicates.

Only matching fields in the jwt, checking auth with tokeninfo is a filter

Reply:
> Only matching fields in the jwt, checking auth with tokeninfo is a filter
You're right. my bad..

Docs updated anyways

Reply:
This looks strange to me too. I expected to see Skipper annotations, rather than `fabric/` ones

Reply:
Skipper annotations would suggest it was an override of the filters. I want to make it clear that this is appending filters to the chain of filters which is applied by the gateway operator

Reply:
Would this not be cleaner:
```
metadata:  name: my-app-gateway  
  annotations:    
    zalando.org/skipper-filter: fadeIn("1m") -> tee("https://some-other-backend.zalandoapis.com/api")
```

Reply:
This only shows a subset of the filters which will end up on the generated ingress though. So I think it would be more confusing